test/circuit.js

```diff
@@ -0,0 +1,290 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const parallel = require('async/parallel')
+const series = require('async/series')
+
+const isNode = require('detect-node')
+
+const utils = require('./utils/circuit')
+
+const proc = utils.setUpProcNode
+const js = utils.setUpJsNode
+const go = utils.setUpGoNode
+
+const ws = utils.wsAddr
+const star = utils.wsStarAddr
+const circuit = utils.circuitAddr
+
+const send = utils.send
+
+const base = '/ip4/127.0.0.1/tcp/0'
+
+const connect = (nodeA, nodeB, relay, timeout, callback) => {
+  if (typeof timeout === 'function') {
+    callback = timeout
+    timeout = 1000
+  }
+
+  series([
+    (cb) => nodeA.ipfsd.api.swarm.connect(ws(relay.addrs), cb),
+    (cb) => setTimeout(cb, timeout),
+    (cb) => nodeB.ipfsd.api.swarm.connect(ws(relay.addrs), cb),
+    (cb) => setTimeout(cb, timeout),
+    (cb) => nodeA.ipfsd.api.swarm.connect(circuit(nodeB.addrs), cb)
+  ], callback)
+}
+
+const timeout = 80 * 1000
+const baseTest = {
+  connect,
+  send,
+  timeout
+}
+
+let tests = {
```

Comment:
Why let and not const?

Reply:
gets merged with the browser tests. I'll separated into two different files as you suggested bellow.